samples/deploy/crd/v1beta1/ApisixRoute.yaml

```
@@ -35,3 +48,146 @@ spec:
     kind: ApisixRoute
     shortNames:
       - ar
+  preserveUnknownFields: false
+  validation:
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          properties:
+            http:
+              type: array
+              minItems: 1
+              items:
+                type: object
+                oneOf:
+                  - required: ["name", "match", "backend"]
+                  - required: ["name", "match", "backends"]
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  priority:
+                    type: integer
+                  match:
+                    type: object
+                    required:
+                    - paths
+                    properties:
+                      paths:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                          pattern: "^/" # start with "/"
+                      hosts:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                          pattern: "^\\*?[0-9a-zA-Z-._]+$"
+                      methods:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                          enum: ["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS", "CONNECT", "TRACE"]
+                      remoteAddrs:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: string
+                      exprs:
+                        type: array
+                        minItems: 1
+                        items:
+                          type: object
+                          properties:
+                            subject:
+                              type: object
+                              properties:
+                                scope:
+                                  type: string
+                                  enum: ["Header", "Query", "Cookie", "Path"]
+                                name:
+                                  type: string
+                                  minLength: 1
+                              required:
+                              - scope
+                            op:
+                              type: string
+                              enum:
+                                - Equal
+                                - NotEqual
+                                - GreaterThan
+                                - LessThan
+                                - In
+                                - NotIn
+                                - RegexMatch
```

So for this enum maybe keep the top part with the custom order and then for the bottom part sort it ?

```suggestion
- RegexMatch
- RegexMatchCaseInsensitive
- RegexNotMatch
- RegexNotMatchCaseInsensitive
```

---

I grouped them by their meanings.